src/app/caesar-portal/caesar-login/service/caesar-login.service.spec.ts

```diff
@@ -0,0 +1,44 @@
+import { inject, TestBed } from '@angular/core/testing';
+import { MockBackend, MockConnection } from '@angular/http/testing';
+
+import { BaseRequestOptions, Http, RequestMethod, Response, ResponseOptions } from '@angular/http';
+import { CaesarLoginService } from './caesar-login.service';
+
+
+describe("CaesarLoginService", () => {
+
+    let subject: CaesarLoginService;
+    let backend: MockBackend;
+
+    beforeEach(() => {
+
+        TestBed.configureTestingModule({
+            providers: [
+                MockBackend,
+                BaseRequestOptions,
+                {
+                    provide: Http,
+                    useFactory: (backendInstance: MockBackend, defaultOptions: BaseRequestOptions) => {
+                        return new Http(backendInstance, defaultOptions);
+                    },
+                    deps: [MockBackend, BaseRequestOptions]
+                },
+                CaesarLoginService
+            ]
+        });
+    });
+
+    beforeEach(inject([CaesarLoginService, MockBackend], (caesarLoginService: CaesarLoginService, mockBackend: MockBackend) => {
+        subject = caesarLoginService;
+        backend = mockBackend;
+    }));
+
+    it("#login should be defined",
+        inject([CaesarLoginService], (service: CaesarLoginService) => {
+            expect(service.login('Vasya', '1234')).toBeDefined();
```

Comment:
I have some trouble with this:
it's service:

export class CaesarLoginService {
    constructor (private http: Http) { 
    }
 
    login (username: string, password: string) {
        const headers = new Headers({
            'Content-Type': 'application/json'
        });

        return this.http.post('/login', JSON.stringify({username: username, password: password}),  {headers: headers});
    } 
_____
And I wrotw this test:

it("#http.post must be called",
        inject([CaesarLoginService], (service: CaesarLoginService) => {
            spyOn(subject, 'http.post');
            subject.login('vasya', '1234');
            expect(subject.http.post).toHaveBeenCalled();
    }));
____
but because of 'http.post' there are a lot of errors in console.